e2e/app_test.go

```diff
@@ -69,36 +69,55 @@ func TestAppRetryCount(t *testing.T) {
 	}
 }
 
-// Checks if the stats are written at the end of each log level output.
+// Checks if the stats are written in necessary conditions.
+// 1. Print with every log level when there is an operation
+// 2. Do not print when used with help & version commands.
 func TestAppDashStat(t *testing.T) {
 	t.Parallel()
 
 	const (
-		bucket      = "bucket"
-		fileContent = "this is a file content"
-		dst         = "."
-		src         = "file1.txt"
+		bucket                  = "bucket"
+		fileContent             = "this is a file content"
+		dst                     = "."
+		src                     = "file1.txt"
+		ExpectedOutputIfPrinted = "Operation\tTotal\tError\tSuccess\t"
 	)
 
 	testcases := []struct {
```

Comment:
Let's change the table test setup. I think it's easier to read the test with 2 fields:

`command` and `expectedOutput`.

```go
testcases := []struct{
	command string
	expectedOutput string
}{
	{
		command: "--stat --log trace cp . s3://bucket/object",
		expectedOutput: expectedOutputIfPrinted,
	},
}
```

Test logic could be:

```go
cmd = s5cmd(strings.Fields(tc.command)...)
result := icmd.RunCmd(cmd)
result.Assert(t, icmd.Success)

out := result.Stdout()
assert.Assert(t, strings.Contains(out, tc.expectedOutput))
```